lib/include/jxl/gain_map.h

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) the JPEG XL Project Authors. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+/** @addtogroup libjxl_metadata
+ * @{
+ * @file gain_map.h
+ * @brief Utility functions to manipulate jhgm (gain map) boxes.
+ */
+
+#ifndef JXL_GAIN_MAP_H_
+#define JXL_GAIN_MAP_H_
+
+#include <jxl/color_encoding.h>
+#include <jxl/jxl_export.h>
+#include <jxl/types.h>
+
+#ifndef __cplusplus
+extern "C" {
+#endif
+
+/** Gain map bundle */
+typedef struct {
+  /** version number */
+  uint8_t jhgm_version;
+  /** size of the gain map metadata */
+  uint16_t gain_map_metadata_size;
+  /** pointer to binary blob of gain map metadata (ISO 21496-1) */
+  const uint8_t* gain_map_metadata;
+  /** indicate if it has a color encoding*/
+  bool has_color_encoding;
+  /** uncompressed color encoding */
+  JxlColorEncoding color_encoding;
+  /** size of the alt_icc profile  (compressed size) */
+  uint32_t alt_icc_size;
+  /** pointer to the compressed icc profile */
+  const uint8_t* alt_icc;
```

Comment:
This means applications have to implement their own ICC profile compression/decompression methods, since this is not something that is currently exposed in the libjxl API. That's not a good idea imo.

Either we should have an _uncompressed_ ICC profile here (and `JxlGainMapWriteBundle` compresses it before writing it, `JxlGainMapReadBundle` decompresses it after writing it), or we should expose functions to compress/decompress ICC profiles. I prefer the first option.

Reply:
We are planning to expose the functions to compress/decompress the ICC profiles